src/jenkins.py

```diff
@@ -144,6 +146,467 @@ def _get_api_credentials(container: ops.Container) -> Credentials:
         raise JenkinsBootstrapError("Admin API credentials not yet setup.") from exc
 
 
+class Jenkins:
```

Comment:
What is the reason please for changing this to be a class?

Reply:
The client and all the method that use it are dependent on the prefix that is set in the pebble plan, as such, the easiest way to track that prefix is to set that prefix as an attribute of the `jenkins` class